pkg/git/git.go

```diff
@@ -0,0 +1,80 @@
+package git
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	giturls "github.com/whilp/git-urls"
+)
+
+func run(args ...string) error {
+	cmd := exec.Command("git", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func HasStagedChange() error {
+	err := run("diff", "--staged", "--exit-code")
+	if err != nil {
+		return errors.Errorf("repository contains staged changes: %q", err)
+	}
+	return nil
+}
+
+func RmRecursive(path string) error {
+	return run("rm", "-r", path)
```

Comment:
whenever I see an `rm -r` I like to see some validation that the path isn't something it shouldn't be.  like `/` :)

Reply:
this line looks scary but `run()` wraps `git` in a way that this command effectively becomes `git rm -r <path>` :)